src/Types/UnbalancedTransaction.purs

```diff
@@ -0,0 +1,69 @@
+module Types.UnbalancedTransaction where
+
+import Data.BigInt (BigInt)
+import Data.Map (Map)
+import Data.Maybe (Maybe)
+import Data.Newtype (class Newtype)
+import Prelude
+import Types.ByteArray (ByteArray(..))
+import Types.POSIXTimeRange (POSIXTimeRange)
+import Types.Transaction (Ed25519KeyHash(..), Transaction, TransactionHash, Value)
+
+newtype PubKey = PubKey ByteArray
+
+derive instance Newtype PubKey _
+derive newtype instance Eq PubKey
+
+newtype PaymentPubKey = PaymentPubKey PubKey
+
+derive instance Newtype PaymentPubKey _
+derive newtype instance Eq PaymentPubKey
+
+newtype TxOutRef = TxOutRef
+  { txOutRefId :: TransactionHash
```

Comment:
This is a stylistic thing, but I'd prefer not to copy the Haskell naming conventions for record fields